sql/testdata/txn

```diff
@@ -152,3 +152,34 @@ COMMIT TRANSACTION
 
 statement error there is no transaction in progress
 ROLLBACK TRANSACTION
+
+# Set isolation level without a transaction is an error.
+
+statement error there is no transaction in progress
+SET TRANSACTION ISOLATION LEVEL SNAPSHOT
+
+statement ok
+BEGIN TRANSACTION ISOLATION LEVEL SNAPSHOT; COMMIT
+
+statement ok
+BEGIN TRANSACTION ISOLATION LEVEL SERIALIZABLE; COMMIT
+
+statement ok
+BEGIN TRANSACTION; SET TRANSACTION ISOLATION LEVEL SNAPSHOT; COMMIT
+
+statement ok
+BEGIN TRANSACTION; SET TRANSACTION ISOLATION LEVEL SERIALIZABLE; COMMIT
+
+# It is an error to change the isolation level of a running transaction.
+
+statement ok
+BEGIN TRANSACTION
+
+statement ok
+UPDATE kv SET v = 'b' WHERE k in ('a')
+
+statement error cannot change the isolation level of a running transaction
+SET TRANSACTION ISOLATION LEVEL SNAPSHOT
+
+statement ok
+ROLLBACK
```

Comment:
Not unless we add a `SHOW TRANSACTION ISOLATION` statement (or something similar - mysql spells it `SHOW VARIABLES LIKE 'tx_isolation'` and postgres has `SELECT current_setting('transaction_isolation')`. Ideally we'd have a more complicated test to engineer the kind of transaction conflict where the isolation level matters, but that will have to be in go instead of a logic_test.